--- ../../pacemaker-1.1.2.1/work/1.1/configure.ac	2010-06-10 21:38:13.451403754 +0200
+++ 1.1/configure.ac	2010-06-13 15:07:11.732285413 +0200
@@ -537,10 +537,7 @@
    AC_MSG_ERROR(You need (g)make installed in order to build ${PACKAGE})
 fi
 
-AM_CONDITIONAL(BUILD_HELP, test x"${HELP2MAN}" != x"")
-if test x"${HELP2MAN}" != x""; then
-   PKG_FEATURES="$PKG_FEATURES manpages"
-fi
+AM_CONDITIONAL(BUILD_HELP, false)
 
 MANPAGE_XSLT=""
 if test x"${XSLTPROC}" != x""; then
