diff -ur cherokee-svn/cherokee/handler_post_report.c cherokee-patched/cherokee/handler_post_report.c
--- cherokee-svn/cherokee/handler_post_report.c	2010-03-21 02:16:09.381787331 +0100
+++ cherokee-patched/cherokee/handler_post_report.c	2010-07-26 12:26:29.241033318 +0200
@@ -192,7 +192,7 @@
 	if (ret != ret_ok) {
 		cherokee_dwriter_dict_open (&hdl->writer);
 		cherokee_dwriter_cstring (&hdl->writer, "error");
-		cherokee_dwriter_cstring (&hdl->writer, state);
+		cherokee_dwriter_string (&hdl->writer, state, strlen(state));
 		cherokee_dwriter_dict_close (&hdl->writer);
 
 		TRACE (ENTRIES, "Post report: '%s'\n", "Could not get info");
