Only in cherokee-patched/cherokee: .collector.c.swp
diff -ur cherokee-svn/cherokee/handler_post_report.c cherokee-patched/cherokee/handler_post_report.c
--- cherokee-svn/cherokee/handler_post_report.c	2010-03-21 02:16:09.381787331 +0100
+++ cherokee-patched/cherokee/handler_post_report.c	2010-07-26 12:26:29.241033318 +0200
@@ -192,7 +192,7 @@
 	if (ret != ret_ok) {
 		cherokee_dwriter_dict_open (&hdl->writer);
 		cherokee_dwriter_cstring (&hdl->writer, "error");
-		cherokee_dwriter_cstring (&hdl->writer, state);
+		cherokee_dwriter_string (&hdl->writer, state, strlen(state));
 		cherokee_dwriter_dict_close (&hdl->writer);
 
 		TRACE (ENTRIES, "Post report: '%s'\n", "Could not get info");
diff -ur cherokee-svn/cherokee/post_track.c cherokee-patched/cherokee/post_track.c
--- cherokee-svn/cherokee/post_track.c	2010-03-21 02:16:09.367787304 +0100
+++ cherokee-patched/cherokee/post_track.c	2010-07-26 15:19:52.965031909 +0200
@@ -194,6 +194,7 @@
 
 	/* Register it
 	 */
+	TRACE (ENTRIES, "test: X-Progress-ID='%s'\n", tmp.buf);
 	CHEROKEE_MUTEX_LOCK (&track->lock);
 
 	ret = cherokee_avl_get (&track->posts_lookup, &tmp, NULL);
@@ -296,6 +297,7 @@
 	/* Properties
 	 */
 	n->last_purge = cherokee_bogonow_now;
+	CHEROKEE_MUTEX_INIT (&n->lock, CHEROKEE_MUTEX_FAST);
 
 	INIT_LIST_HEAD (&n->posts_list);
 
