--- cluster-glue-1.0.2-rc2.old/doc/Makefile.am	2010-01-05 09:53:47.000000000 +0100
+++ cluster-glue-1.0.2-rc2/doc/Makefile.am	2010-01-05 10:07:02.000000000 +0100
@@ -29,7 +29,6 @@
 
 if BUILD_DOC
 man_MANS		+= hb_report.8 ha_logd.8 ha_logger.1 stonith.8 meatclient.8
-endif
 
 EXTRA_DIST		= $(man_MANS)
 
@@ -47,3 +46,4 @@
 	$(XSLTPROC) \
 	$(XSLTPROC_MANPAGES_OPTIONS) \
 	$(MANPAGES_STYLESHEET) $<
+endif
